### PR TITLE
Fix embeddings for PyTorch 1.8

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR fixes several embeddings layer with the recent breaking change introduced in PyTorch 1.8. Up until PyTorch 1.7, the `padding_idx` passed to an embedding layer was used to initialize the corresponding row in the weights to 0 but ignored afterwards.

Now, this `padding_idx` is used at every forward pass and ignores the potential weights of the padding index (spoiler alert, all pretrained models I checked have a nonzero one).

To solve this, this PR removes all `padding_idx` passed to embeddings layer. As we were in any case re-initializing them in the `_init_weights` function, the zero weight for that index was ignored in any case. This PR thus introduces no breaking change on our side while dealing with the breaking change in PyTorch.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Yes, this works for me. Before releasing a new version, I'll need to ensure that *all* slow tests pass.

**Review by LysandreJik — APPROVED** [review]
Yes, this works for me! Good job on solving this tough issue.
